cluster-autoscaler/cloudprovider/azure/azure_container_service_pool.go

```diff
@@ -0,0 +1,435 @@
+		newVal, err := updatedVal.Result(acsClient)
+		if err != nil {
+			glog.Error(err)
+			return err
+		}
+		end = time.Now()
+		glog.Infof("Target size set done, ACS. Val: %+v", newVal)
+	}
+
+	glog.Infof("Got Updated value. Time taken: ", end.Sub(start))
+	return nil
+}
+
+//IncreaseSize calls in the underlying SetSize to increase the size in response
+//to a scale up. It calculates the expected size based on a delta provided as
+//parameter
+func (agentPool *ContainerServiceAgentPool) IncreaseSize(delta int) error {
+	if delta <= 0 {
+		return fmt.Errorf("Size increase must be +ve")
+	}
+	currentSize, err := agentPool.TargetSize()
+	if err != nil {
+		return err
+	}
+	targetSize := int(currentSize) + delta
+	if targetSize > agentPool.MaxSize() {
+		return fmt.Errorf("Size increase request of %d more than max size %d set", targetSize, agentPool.MaxSize())
+	}
+	return agentPool.SetSize(targetSize)
+}
+
+//DeleteNodesInternal calls the underlying vm service to delete the node.
+//Additionally it also call into the container service to reflect the node count resulting
+//from the deletion.
+func (agentPool *ContainerServiceAgentPool) DeleteNodesInternal(providerIDs []string) error {
+	currentSize, err := agentPool.TargetSize()
+	if err != nil {
+		return err
+	}
+	// Set the size to current size. Reduce as we are able to go through.
+	targetSize := currentSize
+
+	for _, providerID := range providerIDs {
+		glog.Infof("ProviderID got to delete: %s", providerID)
+		nodeName, err := agentPool.GetName(providerID)
+		if err != nil {
+			return err
+		}
+		glog.Infof("VM name got to delete: %s", nodeName)
+		ctx, cancel := getContextWithCancel()
+		defer cancel()
+
+		_, err = agentPool.manager.azClient.virtualMachinesClient.Delete(ctx, agentPool.nodeResourceGroup, nodeName)
+		// TODO: Look into the status for more detailed handling.
+		if err != nil {
+			return err
+		}
+		// TODO: ACS requires more cleanup to delete the managed disk
+		targetSize--
+	}
+
+	// TODO: handle the errors from delete operation.
+	if currentSize != targetSize {
+		agentPool.SetSize(targetSize)
+	}
+	return nil
+}
+
+//DeleteNodes extracts the providerIDs from the node spec and calls into the internal
+//delete method.
+func (agentPool *ContainerServiceAgentPool) DeleteNodes(nodes []*apiv1.Node) error {
+	var providerIDs []string
+	for _, node := range nodes {
+		glog.Infof("Node: %s", node.Spec.ProviderID)
+		providerIDs = append(providerIDs, node.Spec.ProviderID)
+	}
+	for _, p := range providerIDs {
+		glog.Infof("ProviderID before calling acsmgr: %s", p)
+	}
+	return agentPool.DeleteNodesInternal(providerIDs)
+}
+
+//GetNodes extracts the node list from the underlying vm service and returns back
+//equivalent providerIDs  as list.
+func (agentPool *ContainerServiceAgentPool) GetNodes() ([]string, error) {
+	ctx, cancel := getContextWithCancel()
+	defer cancel()
+	vmList, err := agentPool.manager.azClient.virtualMachinesClient.List(ctx, agentPool.nodeResourceGroup)
```

Comment:
What if there are other VMs not managed by AKS or ACS in the resouceGroup?

Reply:
Added filter with pool name in the Tags for the vm.